File: 프로그래머스/2/42587. 프로세스/프로세스.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int[] priorities, int location) {
        
        int answer = 0;
        Queue<int[]> queue = new Queue<int[]>();

        for (int i = 0; i < priorities.Length; i++) 
        {
            queue.Enqueue(new int[] { i , priorities[i] });
        }

        int[] target = new int[] { location , priorities[location] };

        while (true) 
        {
            int[] que = queue.Dequeue();
            if (queue.Any(p => p[1] > que[1])) queue.Enqueue(que);
            else 
            {
                answer++;
                if (que.SequenceEqual(target)) return answer; 
            }
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque()
    #for i in range(len(priorities)):
    #    queue.append([i,priorities[i]])
    
    for idx, priority in enumerate(priorities):
        queue.append([idx, priority])
    
    while(queue):
        pick = queue.popleft()
        
        if any(p[1] > pick[1] for p in queue):
            queue.append(pick)
            continue
            
        answer += 1
        
        if pick[0] == location: 
            return answer
    
    return answer